chinglee-iot/FreeRTOS-Kernel
# Merge xTimerGenericCommand from SMP branch

Merge xTimerGenericCommand from SMP branch

Description
-----------
* Split xTimerGenericCommand into xTimerGenericCommandFromTask and xTimerGenericCommandFromISR to remove the recursion path when called from ISRs.
* Add portTIMER_CALLBACK_ATTRIBUTE for timer callback function


Test Steps
-----------
* Run WIN32-MSVC main full
* Run RP2040 main full
* Compile RP2040 SMP main full

Related Issue
-----------


Performance Comparison Test
-----------
* This PR is to support different compiler. No code logic is changed. The performance has no difference.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
